src/conf_mode/system-logs.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import exit
+
+from vyos import ConfigError
+from vyos import airbag
+from vyos.config import Config
+from vyos.logger import syslog
+from vyos.template import render_to_string
+from vyos.util import read_file, write_file
+airbag.enable()
+
+# path to logrotate config for atop
+logrotate_atop_file = '/etc/logrotate.d/vyos-atop'
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base = ['system', 'logs']
+    logs_config = conf.get_config_dict(base)
+
+    return logs_config
+
+
+def verify(logs_config):
+    # Nothing to verify here
+    pass
+
+
+def generate(logs_config):
+    # get configuration for logrotate atop
+    logrotate_atop = logs_config.get('logs', {}).get('logrotate',
```

Comment:
Please use `dict_search('logs.logrotate.atop', logs_config)` instead.

Reply:
`render_to_string()` expect to get an iterable object after. 
So, it is possible to use `dict_search()` here, but we will need then also add another check for `None`/`dict` values, which will make the code even longer. Shall I switch to `dict_search()` despite this?

Reply:
Please do so - then all implementations are more or less lookalikes

Reply:
Replaced to `dict_search()`.